src/helper/Backend.cpp

```diff
@@ -70,6 +75,26 @@ namespace SDDM {
                         .arg(mainConfig.X11.UserAuthFile.get());
                 env.insert(QStringLiteral("XAUTHORITY"), value);
             }
+#if defined(Q_OS_FREEBSD)
+        /* get additional environment variables via setclassenvironment();
+            this needs to be done here instead of in UserSession::setupChildProcess
+            as the environment for execve() is prepared here */
+        login_cap_t *lc;
+
+        if (lc = login_getpwclass(pw)) {
+            setclassenvironment(lc, pw, 1);     /* path variables */
+            setclassenvironment(lc, pw, 0);     /* non-path variables */
+            login_close(lc);
+            if (lc = login_getuserclass(pw)) {
+                setclassenvironment(lc, pw, 1);
+                setclassenvironment(lc, pw, 0);
```

Comment:
To me the documentation suggests that `setusercontext()` is a lower-level function, and I was going to suggest `setusercontext(lc, pw, -1, LOGIN_SETPATH | LOGIN_SETENV)`. But looking in `login_class.c`, it turns out that `setusercontext()` splits out calls into `setclassenvironment()` and others; it's probably worth a one-line comment like
```
// setclassenvironment() is the implementation inside setusercontext()
```